src/net_processing.cpp

```diff
@@ -2605,6 +2609,39 @@ bool SendMessages(CNode* pto, CConnman& connman, std::atomic<bool>& interruptMsg
                 pto->fDisconnect = true;
             }
         }
+        // Check for headers sync timeouts
+        if (state.fSyncStarted && state.nHeadersSyncTimeout < std::numeric_limits<int64_t>::max()) {
+            // Detect whether this is a stalling initial-headers-sync peer
+            if (pindexBestHeader->GetBlockTime() <= GetAdjustedTime() - 24*60*60) {
```

Comment:
I'll add an additional commit to fix this to 6*60*60. Can you explain why these values need to be different in Dash? As I understand it, the change from 24 to 6 would make the approximate amount of blocks (144) to line up with that of Bitcoin again. Does 144 blocks have any other special meaning than being a day for Bitcoin?